### PR TITLE
Handle Failed Decoding Result

### PR DESCRIPTION
**Pull request by rabeyta**
This PR addresses issue #32

### CODE REVIEW

File: riposte-core/src/test/java/com/nike/riposte/server/componenttest/VerifyDecoderFailedResultIsHandledTest.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        );
+
+        //leave off content-length and transfer-encoding to trigger DecoderFailedResult
+        request.headers().set(HttpHeaders.Names.HOST, "localhost");
+
+        // when
+        Pair<Integer, String> serverResponse = executeRequest(request, proxyServerConfig.endpointsPort(), incompleteCallTimeoutMillis);
+
+        // then
+        assertThat(serverResponse.getLeft()).isEqualTo(HttpResponseStatus.BAD_REQUEST.code());
```
Comment (nicmunroe) [inline riposte-core/src/test/java/com/nike/riposte/server/componenttest/VerifyDecoderFailedResultIsHandledTest.java]: This assertion is a little thin for verifying that the server threw the correct exception. Can you do something like [this](https://github.com/Nike-Inc/riposte/blob/f057c68f01fa67c143441de7d6fe4369c9d527b8/riposte-core/src/test/java/com/nike/riposte/server/componenttest/VerifyMiscellaneousFunctionalityComponentTest.java#L65) instead, including error metadata verification?

File: riposte-core/src/test/java/com/nike/riposte/server/componenttest/VerifyDecoderFailedResultIsHandledTest.java [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        );
+
+        //leave off content-length and transfer-encoding to trigger DecoderFailedResult
+        request.headers().set(HttpHeaders.Names.HOST, "localhost");
+
+        // when
+        Pair<Integer, String> serverResponse = executeRequest(request, downstreamServerConfig.endpointsPort(), incompleteCallTimeoutMillis);
+
+        // then
+        assertThat(serverResponse.getLeft()).isEqualTo(HttpResponseStatus.BAD_REQUEST.code());
```
Comment (nicmunroe) [inline riposte-core/src/test/java/com/nike/riposte/server/componenttest/VerifyDecoderFailedResultIsHandledTest.java]: Ditto on error verification.

File: riposte-core/src/test/java/com/nike/riposte/server/componenttest/VerifyDecoderFailedResultIsHandledTest.java (hunk 1 above)
Comment (rabeyta) [inline riposte-core/src/test/java/com/nike/riposte/server/componenttest/VerifyDecoderFailedResultIsHandledTest.java]: Good call. I've added it.

File: riposte-core/src/test/java/com/nike/riposte/server/componenttest/VerifyDecoderFailedResultIsHandledTest.java (hunk 2 above)
Comment (rabeyta) [inline riposte-core/src/test/java/com/nike/riposte/server/componenttest/VerifyDecoderFailedResultIsHandledTest.java]: done!

**Comment by nicmunroe** [comment]
Looks great, thank you!

**Pull request merged**